solutions/projectstaffing/jgraph/core/src/main/scala/com/microsoft/graphdataconnect/skillsfinder/config/MdcFilter.scala

```diff
@@ -19,33 +19,42 @@ import org.springframework.context.annotation.Configuration
 @Configuration
 class MdcFilter(@Autowired override val userService: UserService) extends UserIdExtractionFilter(userService) {
   private val log: Logger = LoggerFactory.getLogger(classOf[MdcFilter])
+  private val arrayAnonymousUserInfo = Map("accesToken" -> "", "idToken" -> "", "refreshToken" -> "", "userId" -> "test@anonymous.com")
 
   @throws[IOException]
   @throws[ServletException]
   override protected def doFilter(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain): Unit = {
-    try {
-      MDC.put("correlationId", getCorrelationId(request))
-      val userId: Option[String] = getUserIdFromHttpServletRequest(request)
 
-      if (userId.isDefined) {
-        MDC.put("user", userId.get)
-      }
+        try {
+          if (anonymousUser) {
+            val userId = arrayAnonymousUserInfo.get("userId")
+            MDC.put("user", userId.get)
```

Comment:
just set the anonymousUserDefaultEmail, instead of calling get on an Option[String]